wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTechnicalReportViewController.swift

```diff
@@ -142,32 +142,45 @@ protocol SendTechnicalReportPresenter: MFMailComposeViewControllerDelegate {
     func presentMailComposer(withLogs logsIncluded: Bool)
 }
 
-
 extension SettingsTechnicalReportViewController: SendTechnicalReportPresenter {}
 
-
 extension SendTechnicalReportPresenter where Self: UIViewController {
     func presentMailComposer(withLogs logsIncluded: Bool) {
         presentMailComposer(withLogs: logsIncluded, sourceView: nil)
     }
 
-    func presentMailComposer(withLogs logsIncluded: Bool, sourceView: UIView?, fallback: ((Bool) -> Void)? = nil) {
+    func presentMailComposer(withLogs logsIncluded: Bool, sourceView: UIView?) {
         let mailRecipient = WireEmail.shared.callingSupportEmail
 
         guard MFMailComposeViewController.canSendMail() else {
-            DebugAlert.displayFallbackActivityController(logPaths: ZMSLog.pathsForExistingLogs, email: mailRecipient, from: self, sourceView: sourceView)
+            DebugAlert.displayFallbackActivityController(logPaths: DebugLogSender.debugLogs, email: mailRecipient, from: self, sourceView: sourceView)
             return
         }
 
         let mailComposeViewController = MFMailComposeViewController()
         mailComposeViewController.mailComposeDelegate = self
         mailComposeViewController.setToRecipients([mailRecipient])
-        mailComposeViewController.setSubject(NSLocalizedString("self.settings.technical_report.mail.subject", comment: ""))
+        mailComposeViewController.setSubject(L10n.Localizable.Self.Settings.TechnicalReport.Mail.subject)
+        mailComposeViewController.setMessageBody("Debug report", isHTML: false)
 
         if logsIncluded {
-            mailComposeViewController.attachLogs()
+            var topMostViewController: SpinnerCapableViewController? = UIApplication.shared.topmostViewController(onlyFullScreen: false) as? SpinnerCapableViewController
+            topMostViewController?.isLoadingViewVisible = true
+
+            Task.detached(priority: .userInitiated, operation: { [topMostViewController] in
+                if #available(iOS 16.0, *) {
+                    try await Task.sleep(for: .seconds(3))
+                } else {
+                    // Fallback on earlier versions
+                }
+                await mailComposeViewController.attachLogs()
+                await MainActor.run {
+                    self.present(mailComposeViewController, animated: true, completion: nil)
+                    topMostViewController?.isLoadingViewVisible = false
+                }
```

Comment:
See other comment about `MainActor.run`

Reply:
here I'll still need it because `topMostViewController?.isLoadingViewVisible = false` this hides the loader an d needs to be run on main

Reply:
Okay